Automation for ocp-cluster-etcd-operator

swapnil-bobade27 opened the pull request:
Automation for ocp-cluster-etcd-operator and kube-burner workload test

swapnil-bobade27 commented on playbooks/ocp-cluster-etcd-operator.yml:
```diff
@@ -0,0 +1,6 @@
+- name: OCP cluster etcd operator verification
+  hosts: bastion
+  roles: 
+  - ocp-cluster-etcd-operator
```
Added

swapnil-bobade27 commented on playbooks/roles/ocp-cluster-etcd-operator/defaults/main.yaml:
```diff
@@ -0,0 +1,6 @@
+etcd_namespace: openshift-etcd
+kube_burner_repo: "https://github.com/kube-burner/kube-burner.git"
+kube_burner_dir: "/root/kube-burner"
+kube_burner_path: "/root/kube-burner/examples/workloads"
```
Added new line

swapnil-bobade27 commented on playbooks/roles/ocp-cluster-etcd-operator/files/api-intensive.yaml:
```diff
@@ -0,0 +1,34 @@
+  register: api_intensive_output
+  failed_when: 
+    - api_intensive_output.rc != 0
+    - "'👋 Exiting kube-burner' not in api_intensive_output.stdout"
```
Added new line to all files

varad-ahirwadkar commented on playbooks/roles/ocp-cluster-etcd-operator/tasks/install-kube-burner.yaml:
```diff
@@ -0,0 +1,26 @@
+    creates: "/tmp/kube-burner-V1.10.4-linux-ppc64le"  # Prevent extraction if the folder already exists
+
+- name: Verify extracted files
+  command: ls -la /tmp/ | grep kube
```
Please grep full name,
```suggestion
  command: ls -la /tmp/ | grep kube-burner
```

varad-ahirwadkar commented on playbooks/roles/ocp-cluster-etcd-operator/README.md:
```diff
@@ -0,0 +1,53 @@
+- It will check the DaemonSet for HEARTBEAT_INTERVAL and LEADER_ELECTION_TIMEOUT parameters
+- Change the control plane hardware speed tolerance to "Slower"
+- Run the acme-air workload and e2e conformance test suite
+- Also it will run the kube-burner workload test
```
Can we create a separate role for kube-burner workload test ? if required we can reuse this role.

varad-ahirwadkar replied:
Can we make it optional to deploy acme-air app and run e2e/kube-burner workload tests ? Or If we are going to run these roles every time, then we can keep it.

swapnil-bobade27 replied:
Added separate role for kube-burner workload and made deploy acme-air app and run e2e/kube-burner workload as optional

varad-ahirwadkar commented on playbooks/main.yml:
```diff
@@ -120,4 +120,8 @@
+
+- import_playbook: ocp-cluster-etcd-operator.yml
+  when: cluster_etcd_enabled is defined and cluster_etcd_enabled
+
```
Please add kube-burner role as well.

swapnil-bobade27 replied:
kube-burner role added

swapnil-bobade27 commented:
> Ideally we should have a separate PR for adding new role ocp-kube-burner.

new role for kube-burner added

varad-ahirwadkar commented:
/lgtm

yussufsh commented on playbooks/roles/ocp-kube-burner/files/cluster-density.yaml:
These looks like some tasks. The files dir is to keep static files.
Please refer to https://docs.ansible.com/ansible/2.8/user_guide/playbooks_best_practices.html#directory-layout

swapnil-bobade27 replied:
Removed files and updated the defaults/main.yaml

yussufsh commented on playbooks/roles/ocp-kube-burner/files/kubelet-density.yaml:
Not sure why so many tasks to add line after line?
Please use template files under `templates` dir.

swapnil-bobade27 replied:
Updated the main.yaml

swapnil-bobade27 replied:
Can we use git patch to implement above?

yussufsh commented:
/cc @aman4433 
The role directory structure is not proper.

aman4433 reviewed (commented):
The PR looks good overall. Please mention the two playbooks in the repository’s `README.md` and clean up the commit message for clarity.

swapnil-bobade27 commented:
> The PR looks good overall. Please mention the two playbooks in the repository’s `README.md` and clean up the commit message for clarity.

Updated the README.md and cleaned up the commit message

aman4433 commented:
/lgtm

SurajGudaji commented on playbooks/roles/ocp-kube-burner/tasks/main.yaml:
```diff
@@ -0,0 +1,161 @@
+    path: "{{ kube_burner_path }}/kubelet-density/templates/pod.yml"
+    insertafter: 'seccompProfile:'
+    line: "        type: RuntimeDefault"
+
```
Can we use blockinfile module to add the above 6 lines?

swapnil-bobade27 replied:
I had tried blockinfile module, but it was not maintaining the indentation

aman4433 commented:
/lgtm

sajauddin commented:
/lgtm

yussufsh reviewed (approved):
/approve
